subprojects/base-services/src/main/java/org/gradle/internal/service/DefaultServiceRegistry.java

```diff
@@ -1156,7 +1156,12 @@ private static void assertValidServiceType(Class<?> serviceClass) {
     private static String format(Type type) {
         if (type instanceof Class) {
             Class<?> aClass = (Class) type;
-            return aClass.getSimpleName();
+            Class<?> enclosingClass = aClass.getEnclosingClass();
```

Comment:
* what's the reason for using `.` and not `$` as a delimiter?
* would it make more sense to use the `aClass.getName()` instead?

Reply:
I was going for package naming conventions. I thought that `$` was for anonymous inner classes.

I'm happy with either.

I was going for `TextUrlResourceLoader.Factory` to be printed in the error. But I guess `TextUrlResourceLoader$Factory` could also make sense.

Which would you rather see?

Reply:
Personally, the simpler one, that requires less custom code. Don't mind either way, just proposed an alternative.

Reply:
Fixed to use `$`.